library/core/src/fmt/mod.rs

```diff
@@ -94,15 +94,12 @@ pub struct Error;
 
 /// A collection of methods that are required to format a message into a stream.
 ///
-/// This trait is the type which this modules requires when formatting
-/// information. This is similar to the standard library's [`io::Write`] trait,
-/// but it is only intended for use in libcore.
+/// This trait is used by this module for formatting information.
+/// This is similar to the standard library's [`io::Write`] trait, but it only
+/// works for UTF-8 encoded data. If you only want to accept Unicode, you can
+/// implement this trait, but if you want to be able to accept raw bytes too,
+/// you should use [`io::Write`].
```

Comment:
Is this clear enough?

Reply:
The io::Write trait is closer to a superset of the fmt::Write trait, because both have write_fmt, and write_str is equivalent to write_all, with write_char the only odd one out. io::Write in addition also has flush management. This should express that it is not similar, _but rather_ you should probably implement one or the other, based on 1) whether it is UTF8 data and 2) whether you need io::Write's flush capability.

It should still mention `write!` being generalized over both Write traits, as well.